File: Top Interview 150/Java Code/Hashmap/128-longest-consecutive-sequence.java
//Time :O(n)
//Space:O(n)
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length==0) return 0;
        Set<Integer>s=new TreeSet<>();
        for(int i=0;i<nums.length;i++) s.add(nums[i]);
        int best=1,count=1;
        for(int it:s){
            int k=it+1;
            if(s.contains(k)){
                count++;
                best=Math.max(best,count);
            }
            else count=1;
        }
        return best;
    }
}

File: Top Interview 150/C++ Code/Hashmap/128-longest-consecutive-sequence.cpp
//Time :O(n)
//Space:O(n)
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
int n=nums.size();
        if(n==0) return 0;
        int best=1,count=1;
        set<int>s;
        for(int i=0;i<n;i++) s.insert(nums[i]);
        for(auto it:s){
            int i=it+1;
            if(s.count(i)){
                count++;
                best=max(best,count);
            }
            else count=1;
        }
        return best;
    }
};
